contrib/babelfishpg_tsql/src/backend_parser/gram-tsql-epilogue.y.c

```diff
@@ -590,6 +590,52 @@ int getElemTypMod(TypeName *t)
     }
 }
 
+Node*
+TsqlJsonModifyMakeFuncCall(Node* expr, Node* path, Node* newValue)
+{
+	FuncCall* fc;
+	List* func_args = list_make3(expr, path, newValue);
+	if(IsA(newValue, FuncCall))
+	{
+		FuncCall* fc_newval = (FuncCall*) newValue;
+		if(is_json_modify(fc_newval->funcname) || is_json_query(fc_newval->funcname))
+		{
+			fc = makeFuncCall(TsqlSystemFuncName("json_modify_no_escape"), func_args, COERCE_EXPLICIT_CALL, -1);
+		}
+		else
+			fc = makeFuncCall(TsqlSystemFuncName("json_modify"), func_args, COERCE_EXPLICIT_CALL, -1);
+	}
+	else
+		fc = makeFuncCall(TsqlSystemFuncName("json_modify"), func_args, COERCE_EXPLICIT_CALL, -1);
+	return (Node*) fc;
```

Comment:
Minor nit: is it possible to also add an alias (if there is no existing alias) so that the output from the call always looks like json_modify (instead of json_modify_no_escape)? You may also need to adjust the analyzer to match this behavior.